sway-lib-std/src/storage.sw

```diff
@@ -106,3 +109,189 @@ impl<K, V> StorageMap<K, V> {
         get::<V>(key)
     }
 }
+
+/// A persistant vector struct
+pub struct StorageVec<V> {}
+
+impl<V> StorageVec<V> {
+    /// Appends the value to the end of the vector
+    /// 
+    /// # Arguments
+    /// 
+    /// * `value` - The item being added to the end of the vector
+    #[storage(read, write)]
+    pub fn push(self, value: V) {
+        // The length of the vec is stored in the __get_storage_key() slot
+        let len = get::<u64>(__get_storage_key());
+        
+        // Storing the value at the current length index (if this is the first item, starts off at 0)
+        let key = sha256((len, __get_storage_key()));
+        store::<V>(key, value);
+
+        // Incrementing the length
+        store(__get_storage_key(), len + 1);
+    }
+
+    /// Removes the last element of the vector and returns it, None if empty
+    #[storage(read, write)]
+    pub fn pop(self) -> Option<V> {
+        let len = get::<u64>(__get_storage_key());
+        // if the length is 0, there is no item to pop from the vec
+        if len == 0 {
+            return Option::None;
+        }
+    
+        // reduces len by 1, effectively removing the last item in the vec
+        store(__get_storage_key(), len - 1);
+
+        let key = sha256((len - 1, __get_storage_key()));
```

Comment:
Can you also abstract out getting/setting an element into a helper function (i.e. the internal getting and setting, with no safety checks)? As-is, you're computing `sha256((position, __get_storage_key()))` as the storage slot several times

Reply:
`__get_storage_key()` will only work inside a struct, so not possible, as calling self functions isnt supported